### PR TITLE
[Merged by Bors] - feat(CategoryTheory/Sites): the property of being a sheaf of types is invariant under natural equivalences

### PR DESCRIPTION
**Pull request by joelriou**
If `P₁ : Cᵒᵖ ⥤ Type w` and `P₂ : Cᵒᵖ ⥤ Type w'` are naturally equivalent presheaves of types (in possibly different universes), then `P₁` is a sheaf for a Grothendieck topology iff `P₂` is.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/Sites/IsSheafFor.lean [hunk 1]
```diff
@@ -603,21 +603,41 @@ theorem isSheafFor_top_sieve (P : Cᵒᵖ ⥤ Type w) : IsSheafFor P ((⊤ : Sie
   rw [← isSheafFor_iff_generate]
   apply isSheafFor_singleton_iso
 
+/-- If `P₁ : Cᵒᵖ ⥤ Type w` and `P₂  : Cᵒᵖ ⥤ Type w` are two naturally equivalent
+presheaves, and `P₁` is a sheaf for a presieve `R`, then `P₂` is also a sheaf for `R`. -/
+lemma isSheafFor_of_nat_equiv {P₁ : Cᵒᵖ ⥤ Type w} {P₂ : Cᵒᵖ ⥤ Type w'}
+    (e : ∀ ⦃X : C⦄, P₁.obj (op X) ≃ P₂.obj (op X))
+    (he : ∀ ⦃X Y : C⦄ (f : X ⟶ Y) (x : P₁.obj (op Y)),
+    e (P₁.map f.op x) = P₂.map f.op (e x))
```
Comment (dagurtomas) [inline Mathlib/CategoryTheory/Sites/IsSheafFor.lean:611]:
```suggestion
    (he : ∀ ⦃X Y : C⦄ (f : X ⟶ Y) (x : P₁.obj (op Y)),
      e (P₁.map f.op x) = P₂.map f.op (e x))
```

I find this slightly more readable

**Comment by dagurtomas** [comment]
Thanks!

maintainer delegate

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors d+

**Comment by joelriou** [comment]
Thanks!

bors merge

**Pull request closed without merging**